src/main/Product.java

```diff
@@ -0,0 +1,33 @@
+package main;
+public class Product {
+    private final String name;
+    private final Euro price;
+
+    public Product(String name, Euro price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Euro getPrice() {
+        return price;
+    }
+
+    @Override
+    public String toString() {
+        return getName() + ";" + getPrice();
```

Comment:
```suggestion
        return name + ";" + price;
```
call public non final getters in own class is antipattern.